source/d3d11/d3d11_device_context.hpp

```diff
@@ -191,14 +194,27 @@ struct D3D11DeviceContext : ID3D11DeviceContext3
 	#pragma endregion
 
 	void log_drawcall(UINT vertices);
+	ID3D11DepthStencilView *copy_depthstencil(ID3D11DepthStencilView *depthstencil);
+	com_ptr<ID3D11Texture2D> copy_depth_texture(D3D11_TEXTURE2D_DESC texture_desc, ID3D11Texture2D *depthtexture);
+	bool check_depthstencil(ID3D11DepthStencilView* pDepthStencilView, com_ptr<ID3D11Texture2D> texture, D3D11_TEXTURE2D_DESC texture_desc);
 	void set_active_depthstencil(ID3D11DepthStencilView* pDepthStencilView);
+	void set_active_OM_depthstencil(ID3D11DepthStencilView* pDepthStencilView);
+	void set_active_cleared_depthstencil(ID3D11DepthStencilView* pDepthStencilView);
 
 	void clear_drawcall_stats();
+	void screen_dimensions();
 
 	LONG _ref = 1;
 	ID3D11DeviceContext *_orig;
 	unsigned int _interface_version;
 	D3D11Device *const _device;
 	com_ptr<ID3D11DepthStencilView> _active_depthstencil;
+	com_ptr<ID3D11Texture2D> _depth_texture;
 	reshade::d3d11::draw_call_tracker _draw_call_tracker;
+	FLOAT _screen_width = 0;
+	FLOAT _screen_height = 0;
+	std::atomic<unsigned int> _clear_DSV_iter = 1;
```

Comment:
If this still works with these variables being local to a device context (not static), then `std::atomic` isn't actually needed anymore, because all operations local to one device context have to run on the same thread, so no race conditions can occur. Atomic operations or synchronization is only necessary if multiple device contexts could access the same data.

Reply:
Yes, but i want those variables to be device context independant. It seems that some games use multiple device contexts in parallel, but maybe i'am wrong...

Reply:
It is in the case of deferred context (as ClearDepthStencilView method can be embedded in a command list, so i don't know the order of execution)...

Reply:
That is all good and well, I just noted, that the way they are used right now means atomics are not needed. Had they been static, this would be an entirely different story, but in you code right now they are local to a device context and therefore to a thread, so no atomics necessary.

Reply:
Thoughts?

Reply:
Thks Crosire for your suggestions. I have to test it in order to see if it will not break the depth buffer detection, especially when the user force the number of the clear pass he wants reshade to retrieve the depth data